app/platform/__init__.py

```diff
@@ -373,3 +378,155 @@ def edit_form(grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id:
+        grant=form.section.collection_schema.grant,
+        collection=form.section.collection_schema,
+        section=form.section,
+        db_form=form,
+        chosen_question_data_type=question_data_type_enum,
+        form=wt_form,
+    )
+
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/questions/<uuid:question_id>/move/<string:direction>",
+    methods=["POST"],
+)
+@mhclg_login_required
+@auto_commit_after_request
+def move_question(
+    grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id: UUID4, question_id: UUID4, direction: str
+) -> ResponseReturnValue:
+    question = get_question_by_id(question_id=question_id)
+
+    if direction == "up":
+        move_question_up(question)
+    elif direction == "down":
+        move_question_down(question)
+
+    return redirect(
+        url_for(
+            "platform.manage_form",
+            grant_id=grant_id,
+            collection_id=collection_id,
+            section_id=section_id,
+            form_id=form_id,
+            back_link="manage_section",
+        )
+    )
+
+
+@platform_blueprint.route(
+    "/grants/<uuid:grant_id>/developers/collections/<uuid:collection_id>/sections/<uuid:section_id>/forms/<uuid:form_id>/questions/<uuid:question_id>/edit",
+    methods=["GET", "POST"],
+)
+@mhclg_login_required
+@auto_commit_after_request
+def edit_question(
+    grant_id: UUID4, collection_id: UUID4, section_id: UUID4, form_id: UUID4, question_id: UUID4
+) -> ResponseReturnValue:
+    question = get_question_by_id(question_id=question_id)
+    wt_form = QuestionForm(obj=question)
+    if wt_form.validate_on_submit():
+        try:
+            assert wt_form.text.data is not None
+            assert wt_form.hint.data is not None
+            assert wt_form.question_data_type.data is not None
+            assert wt_form.name.data is not None
+            update_question(question=question, text=wt_form.text.data, hint=wt_form.hint.data, name=wt_form.name.data)
+            return redirect(
+                url_for(
+                    "platform.manage_form",
+                    grant_id=grant_id,
+                    collection_id=collection_id,
+                    section_id=section_id,
+                    form_id=form_id,
+                    back_link="manage_section",
+                )
+            )
+        except DuplicateValueError as e:
+            field_with_error: Field = getattr(wt_form, e.field_name)
+            field_with_error.errors.append(f"{field_with_error.label.text} already in use")  # type:ignore[attr-defined]
+
+    return render_template(
+        "platform/developers/edit_question.html",
+        grant=question.form.section.collection_schema.grant,
+        collection=question.form.section.collection_schema,
+        section=question.form.section,
+        form=question.form,
+        question=question,
+        wt_form=wt_form,
```

Comment:
```suggestion
        db_form=question.form,
        question=question,
        form=wt_form,
```

I added a meta test I hoped would catch this but I clearly did it wrong - can you fix the validation please? 😓 https://github.com/communitiesuk/funding-service/blob/1cc47d972ac39e625a59dcdb86640e91bac00286/tests/integration/conftest.py#L98

Reply:
Ah, nothing wrong with the test, but you have to do something that actually triggers the template with an error, ie have a test for all the templates 🫣